src/transport/SecureSessionTable.cpp

```diff
@@ -99,32 +105,59 @@ SecureSession * SecureSessionTable::EvictAndAllocate(uint16_t localSessionId, Se
 
     auto cleanup = MakeDefer([this]() { mRunningEvictionLogic = false; });
 
+#if CONFIG_BUILD_FOR_HOST_UNIT_TEST
+    uint32_t maxSessionTableSize = mMaxSessionTableSize;
+#else
+    uint32_t maxSessionTableSize = CHIP_CONFIG_SECURE_SESSION_POOL_SIZE;
+#endif
+
     ChipLogProgress(SecureChannel, "Evicting a slot for session with LSID: %d, type: %u", localSessionId,
                     (uint8_t) secureSessionType);
 
-    VerifyOrDie(mEntries.Allocated() <= CHIP_CONFIG_SECURE_SESSION_POOL_SIZE);
+    VerifyOrDie(mEntries.Allocated() <= maxSessionTableSize);
 
     //
     // Create a temporary list of objects each of which points to a session in the existing
     // session table, but are swappable. This allows them to then be used with a sorting algorithm
     // without affecting the sessions in the table itself.
     //
-    Platform::ScopedMemoryBufferWithSize<SortableSession> sortableSessions;
-    sortableSessions.Calloc(mEntries.Allocated());
-    if (!sortableSessions)
-    {
-        VerifyOrDieWithMsg(false, SecureChannel, "We couldn't allocate a session!");
-        return nullptr;
-    }
+    SortableSession sortableSessions[CHIP_CONFIG_SECURE_SESSION_POOL_SIZE];
```

Comment:
How big is this array? I imagine previous logic allocated on the heap to avoid too much stack usage.

Reply:
Maybe add a comment here so someone else reading code directly understand reason why this is on the stack?

Reply:
Good point. Will add that extra comment with information.